### PR TITLE
PT-33: added "read/<rule_id>" endpoint to update page_end

### PR DESCRIPTION
**Pull request by jmun10**

### CODE REVIEW

**Comment by michardy** [comment]
Is there a reason that is a put request?

**Comment by jmun10** [comment]
@michardy http protocol states that PUT requests should be used for updating entities. This method locates and updates a Pageload entry if it exists

File: app/ptracker.py [hunk 1]
```diff
@@ -143,5 +143,42 @@ def get_pageloads_per_company():
...
+        session.close()
+        return Response(status=404)
+
+    # if date is already filled, do not update
+    if record.page_end is not None:
+        session.close()
+        return Response(status=400)
+
+    record.page_end = datetime.utcnow()
+    session.add(record)
```
Comment (michardy) [inline app/ptracker.py:169]: I am pretty sure that the `session.add` line can be removed.  My reading of the [documentation](https://docs.sqlalchemy.org/en/13/orm/session_api.html#sqlalchemy.orm.session.Session.add) seems to indicate that it should silently fail due to double adding an object.  

Having said all that the code does work

**Review by michardy — APPROVED** [review]

**Comment by timle3** [comment]
Approving since Michael checked it out and says it works.

**Review by timle3 — APPROVED** [review]

File: app/ptracker.py (hunk 1 above)
Comment (jmun10) [reply app/ptracker.py:169]: From what I understand `session.add` is necessary. Code seems to work as it is so i will leave it as is and merge

**Pull request merged**